### PR TITLE
Add gradient accumulation option

### PR DESCRIPTION
**Pull request by andreasHovaldt**
We encountered memory limitations when pretraining the Sonata model, which forced us to significantly reduce the batch size. To address this, we added support for gradient accumulation, enabling us to train with default hyperparameters on limited GPU memory.

The feature is controlled by the ```gradient_accumulation_steps``` config parameter. When set to ```1``` (the default, as added in the base config), the behavior is identical to before, ensuring backward compatibility.

### CODE REVIEW

**Comment by Gofinge** [comment]
@andreasHovaldt Hey, thanks for the contribution. I have one concern regarding the scheduler total step and made the following modification: [f361733](https://github.com/Pointcept/Pointcept/pull/493/commits/f361733011bc2752bc3dd10fa83fd1e3a529f69a)

Could you help me varify whether my change is correct by checking wandb or tensorboard learning rate curve?

**Comment by andreasHovaldt** [comment]
@Gofinge Good catch! I ran a short test with ```gradient_accumulation_steps=4``` and as you can see on the wandb screenshot, run 2 with your addition looks much better than run 1.

image

**Comment by Gofinge** [comment]
Thanks! I will merge the PR. Thank you again for your contribution.
